python/cog/predictor.py

```diff
@@ -306,21 +306,26 @@ def get_input_create_model_kwargs(signature: inspect.Signature) -> Dict[str, Any
         if parameter.default is inspect.Signature.empty:
             default = Input()
         else:
-            default = parameter.default
             # If user hasn't used `Input`, then wrap it in that
-            if not isinstance(default, FieldInfo):
-                default = Input(default=default)
+            if not isinstance(parameter.default, FieldInfo):
+                default = Input(default=parameter.default)
+            else:
+                default = parameter.default
 
         # Fields aren't ordered, so use this pattern to ensure defined order
         # https://github.com/go-openapi/spec/pull/116
-        default.extra["x-order"] = order
+        if PYDANTIC_V2:
+            extra = default.json_schema_extra = {}
```

Comment:
🤨 

```suggestion
            extra = {}
```

Reply:
no, then it wouldn't be set on the default field. the problem is that default.json_schema_extra is None by default, not {} like default.extra in v1